xarray/tests/test_indexes.py

```diff
@@ -697,3 +697,10 @@ def test_safe_cast_to_index_datetime_datetime():
     actual = safe_cast_to_index(np.array(dates))
     assert_array_equal(expected, actual)
     assert isinstance(actual, pd.Index)
+
+
+@pytest.mark.parametrize("dtype", ["int32", "float32"])
+def test_restore_dtype_on_multiindexes(dtype):
```

Comment:
```suggestion
def test_restore_dtype_on_multiindexes(dtype) -> None:
```
This makes sure to test the typing in the test as well.